service/src/intTest/java/uk/nhs/adaptors/gp2gp/gpc/GetGpcStructuredComponentTest.java

```diff
@@ -0,0 +1,137 @@
+
+        GetGpcStructuredTaskDefinition structuredTaskDefinition = buildValidStructuredTask(ehrExtractStatus);
+        getGpcStructuredTaskExecutor.execute(structuredTaskDefinition);
+
+        var ehrExtractUpdated = ehrExtractStatusRepository.findByConversationId(ehrExtractStatus.getConversationId()).get();
+        assertThatInitialRecordWasUpdated(ehrExtractUpdated, ehrExtractStatus);
+
+        var inputStream = storageConnector.downloadFromStorage(ehrExtractStatus.getConversationId() + GPC_STRUCTURED_FILE_EXTENSION);
+        String storageDataWrapperString = IOUtils.toString(inputStream, StandardCharsets.UTF_8);
+
+        var storageDataWrapper = OBJECT_MAPPER.readValue(storageDataWrapperString, StorageDataWrapper.class);
+        assertThatObjectCreated(storageDataWrapper, ehrExtractUpdated, structuredTaskDefinition);
+    }
+
+    private GetGpcStructuredTaskDefinition buildValidStructuredTask(EhrExtractStatus ehrExtractStatus) {
+        return GetGpcStructuredTaskDefinition.builder()
+            .fromAsid(ehrExtractStatus.getEhrRequest().getFromAsid())
+            .toAsid(ehrExtractStatus.getEhrRequest().getToAsid())
+            .fromOdsCode(ehrExtractStatus.getEhrRequest().getFromOdsCode())
+            .conversationId(ehrExtractStatus.getConversationId())
+            .nhsNumber(ehrExtractStatus.getEhrRequest().getNhsNumber())
+            .requestId(ehrExtractStatus.getEhrRequest().getRequestId())
+            .taskId(UUID.randomUUID().toString())
+            .build();
+    }
+
+    private void assertThatInitialRecordWasUpdated(EhrExtractStatus ehrExtractStatusUpdated, EhrExtractStatus ehrExtractStatus) {
+        assertThat(ehrExtractStatusUpdated.getUpdatedAt()).isNotEqualTo(ehrExtractStatus.getUpdatedAt());
+        var gpcAccessStructured = ehrExtractStatusUpdated.getGpcAccessStructured();
+        assertThat(gpcAccessStructured.getObjectName()).isEqualTo(ehrExtractStatus.getConversationId() + GPC_STRUCTURED_FILE_EXTENSION);
+        assertThat(gpcAccessStructured.getAccessedAt()).isNotNull();
+        assertThat(gpcAccessStructured.getTaskId()).isNotNull();
+    }
+
+    private void assertThatObjectCreated(StorageDataWrapper storageDataWrapper, EhrExtractStatus ehrExtractStatus,
+        GetGpcStructuredTaskDefinition structuredTaskDefinition) {
+        assertThat(storageDataWrapper.getConversationId()).isEqualTo(ehrExtractStatus.getConversationId());
+        assertThat(storageDataWrapper.getTaskId()).isEqualTo(ehrExtractStatus.getGpcAccessStructured().getTaskId());
+        assertThat(storageDataWrapper.getType()).isEqualTo(structuredTaskDefinition.getTaskType().getTaskTypeHeaderValue());
+        assertThat(storageDataWrapper.getResponse()).isNotBlank();
+    }
+
+    @Test
+    public void When_StructuredTaskRunTwice_Expect_ObjectToBeOverwrittenInStroage() throws IOException {
+        var ehrExtractStatus = EhrExtractStatusTestUtils.prepareEhrExtractStatus();
```

Comment:
Test methods reuse similar peaces of code. Could we move these pieces to separate private methods?